### DISCUSSION TITLE
it Works with prerecorded file, but get 403 error in streaming connection build

### DISCUSSION THREAD

**Discussion by tanyinyan** [deepgram/community#270] [General help] [answered]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

when running code bellow, the output is
🔴 ERROR: Could not connect to Deepgram! None
🔴 Please contact Deepgram Support with request ID None

sync def main():
    try:
        async with websockets.connect('wss://api.deepgram.com/v1/listen',
        # Remember to replace the YOUR_DEEPGRAM_API_KEY placeholder with your Deepgram API Key
        extra_headers = { 'Authorization': f'token '
                                           f'***********'}) as ws:
            # If the request is successful, print the request ID from the HTTP header
            print('🟢 Successfully opened connection')
            print(f'Request ID: {ws.response_headers["dg-request-id"]}')
            await ws.send(json.dumps({
                'type': 'CloseStream'
            }))
    except websockets.exceptions.InvalidStatusCode as e:
        # If the request fails, print both the error message and the request ID from the HTTP headers
        print(f'🔴 ERROR: Could not connect to Deepgram! {e.headers.get("dg-error")}')
        print(f'🔴 Please contact Deepgram Support with request ID {e.headers.get("dg-request-id")}')

asyncio.run(main())

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

wss://api.deepgram.com/v1/listen

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

No request ID returned

**Comment by jjmaldonis** [comment]
Hey @tanyinyan this almost always means there is something wrong with your Deepgram API key. Did you set it in the code? If so, make sure it's the correct value. It should not have any dashes in it.

**Reply by tanyinyan** [reply]
I tried in another place without proxy, it works. so is maybe the issue of proxy.
Is there a way to disable http(s) proxy in python code?

**Reply by jjmaldonis** [reply, marked as answer]
Here is a link I found about how to disable https proxy in python code: https://stackoverflow.com/a/28521696

However, if you are sending a URL to Deepgram that is `http` rather than `https`, then Deepgram is accessing the URL rather than your own computer. If that's the case, the URL may need to be changed to an `https` URL. Is this possible for you to do?
